daemon/cmd/config.go

```diff
@@ -153,7 +154,8 @@ func getConfigHandler(d *Daemon, params GetConfigParams) middleware.Responder {
 	option.Config.ConfigPatchMutex.RUnlock()
 
 	// Manually add fields that are behind accessors.
-	m["Devices"] = option.Config.GetDevices()
+	devs, _ := tables.SelectedDevices(d.devices, d.db.ReadTxn())
```

Comment:
Fetching device names seems to be repeated thru out this pr. ie get selected devices -> extract device names. Perhaps best to make it a convenience call aka `tables.SelectedDeviceNames`?

Reply:
I want to avoid encouraging dumbing down the device structs to strings as much as possible - it's as is to avoid blowing up this PR more. Subsequent PRs will transform more of the downstream code to take a device struct, so then this will go away.